platform/posix/transport/src/mbedtls_pkcs11_posix.c

```diff
@@ -0,0 +1,958 @@
+ *
+ * @return #MBEDTLS_PKCS11_SUCCESS, #MBEDTLS_PKCS11_INSUFFICIENT_MEMORY, #MBEDTLS_PKCS11_INVALID_CREDENTIALS,
+ * #MBEDTLS_PKCS11_HANDSHAKE_FAILED, or #MBEDTLS_PKCS11_INTERNAL_ERROR.
+ */
+static MbedtlsPkcs11Status_t configureMbedtls( MbedtlsPkcs11Context_t * pMbedtlsPkcs11Context,
+                                               const char * pHostName,
+                                               const MbedtlsPkcs11Credentials_t * pMbedtlsPkcs11Credentials,
+                                               uint32_t recvTimeoutMs );
+
+/**
+ * @brief Configure the credentials in the MbedTLS SSL context.
+ *
+ * @param[in] pMbedtlsPkcs11Context Network context.
+ * @param[in] pMbedtlsPkcs11Credentials TLS setup parameters.
+ *
+ * @return #MBEDTLS_PKCS11_SUCCESS on success,
+ * #MBEDTLS_PKCS11_INVALID_CREDENTIALS on error.
+ */
+static MbedtlsPkcs11Status_t configureMbedtlsCertificates( MbedtlsPkcs11Context_t * pMbedtlsPkcs11Context,
+                                                           const MbedtlsPkcs11Credentials_t * pMbedtlsPkcs11Credentials );
+
+/**
+ * @brief Configure the SNI and ALPN in the MbedTLS SSL context.
+ *
+ * @param[in] pMbedtlsPkcs11Context Network context.
+ * @param[in] pMbedtlsPkcs11Credentials TLS setup parameters.
+ * @param[in] pHostName Remote host name, used for server name indication.
+ *
+ * @return #MBEDTLS_PKCS11_SUCCESS on success,
+ * #MBEDTLS_PKCS11_INVALID_CREDENTIALS on error.
+ */
+static MbedtlsPkcs11Status_t configureMbedtlsSniAlpn( MbedtlsPkcs11Context_t * pMbedtlsPkcs11Context,
+                                                      const MbedtlsPkcs11Credentials_t * pMbedtlsPkcs11Credentials,
+                                                      const char * pHostName );
+
+/**
+ * @brief Configure the SNI and Maximum Fragment Length in the MbedTLS SSL context.
+ *
+ * @param[in] pMbedtlsPkcs11Context Network context.
+ *
+ * @return #MBEDTLS_PKCS11_SUCCESS on success,
+ * #MBEDTLS_PKCS11_INVALID_CREDENTIALS on error.
+ */
+static MbedtlsPkcs11Status_t configureMbedtlsFragmentLength( MbedtlsPkcs11Context_t * pMbedtlsPkcs11Context );
+
+/**
+ * @brief Callback that wraps PKCS #11 for pseudo-random number generation.
+ *
+ * @param[in] pCtx Caller context.
+ * @param[in] pRandom Byte array to fill with random data.
+ * @param[in] randomLength Length of byte array.
+ *
+ * @return Zero on success.
+ */
+static int32_t generateRandomBytes( void * pCtx,
+                                    unsigned char * pRandom,
+                                    size_t randomLength );
+
+/**
+ * @brief Helper for reading the specified certificate object, if present,
+ * out of storage, into RAM, and then into an mbedTLS certificate context
+ * object.
+ *
+ * @param[in] pContext Caller TLS context.
+ * @param[in] pLabelName PKCS #11 certificate object label.
+ * @param[in] class PKCS #11 certificate object class.
+ * @param[out] pCertificateContext Certificate context.
+ *
+ * @return Zero on success.
```

Comment:
Should this function return a `bool` type if only returns binary information of success or failure?